src/tests/test_engine.rs

```diff
@@ -328,13 +328,21 @@ fn default_value_constant() -> TestResult {
 }
 
 #[test]
-fn default_value_not_constant() -> TestResult {
+fn default_value_not_constant1() -> TestResult {
     fail_test(
         r#"def foo [x = ("foo" | str length)] { $x }; foo"#,
         "expected a constant",
     )
 }
 
+#[test]
+fn default_value_not_constant2() -> TestResult {
+    fail_test(
+        r#"def foo [--x = ("foo" | str length)] { $x }; foo"#,
```

Comment:
I think we could omit the call to make sure the error is already emitted on definition or is there a limitation?

Reply:
I can still make the change if you want (but idk if it merits a separate PR).